Storage/SQL/SQLiteHistory.swift

```diff
@@ -82,18 +82,29 @@ public class SQLiteHistory {
 
     private var ignoredSchemes = ["about"]
 
-    required public init(db: BrowserDB) {
+    required public init?(db: BrowserDB, version: Int? = nil) {
         self.db = db
         self.favicons = FaviconsTable<Favicon>()
-        db.createOrUpdate(self.favicons)
+        if !db.createOrUpdate(self.favicons) {
+            return nil
+        }
 
         // BrowserTable exists only to perform create/update etc. operations -- it's not
         // a queryable thing that needs to stick around.
-        db.createOrUpdate(BrowserTable())
+        if !db.createOrUpdate(BrowserTable(version: version ?? BrowserTable.DefaultVersion)) {
+            return nil
+        }
     }
 }
 
 extension SQLiteHistory: BrowserHistory {
+    public func removeSiteFromTopSites(site: Site) -> Success {
+        if let domain = site.domain {
+            return db.run([("UPDATE \(TableDomains) set showOnTopSites = 0 WHERE domain = ?", [domain])])
+        }
+        return deferResult(DatabaseError(description: "Invalid url for site \(site.url)"))
```

Comment:
Why not just `return succeed()`? After all, if the URL is missing, it won't show on top sites…